### PR TITLE
Support invoking node e2e tests from Prow

### PR DESCRIPTION
**Pull request by yguo0905**
Ref: https://github.com/kubernetes/test-infra/issues/2632

- Add a "local" mode to `kubernetes_kubelet.py`. Prow does not support docker run, we will need to trigger the tests locally.
- Add a test job `ci-kubernetes-e2e-gce-ubuntu-1-7-node-prow` for testing invoking node e2e tests from Prow.
- Copy https://github.com/kubernetes/kubernetes/blob/master/test/e2e_node/jenkins/e2e-node-jenkins.sh and https://github.com/kubernetes/kubernetes/blob/master/test/e2e_node/jenkins/jenkins-ci-ubuntu.properties to `jobs/e2e-node`.
- New prow-e2e image is needed since it's based on golang:1.8.3.
- The changes in `jenkins/bootstrap_test.py` is temporary. Once we are able to use Prow to trigger the tests, we will work on making the node e2e test configs be consistent with others.

/assign @krzyzacy 
/cc @abgworrall

### CODE REVIEW

File: scenarios/kubernetes_kubelet.py [hunk 1]
```diff
@@ -95,6 +96,56 @@ def main(script, properties, branch, ssh, ssh_pub, robot):
...
+
+
+def run_local_mode(script, properties, branch, private_key, public_key):
+    """Checkout, build and trigger the node e2e tests locally."""
+    root = os.getcwd()
+    gopath = os.path.join(root, 'go')
+    os.makedirs(os.path.join(gopath, 'src/k8s.io'))
+    k8s = os.path.join(gopath, 'src/k8s.io', 'kubernetes')
+    check(
+        'git', 'clone', '--depth=1',
```
Comment (krzyzacy) [inline scenarios/kubernetes_kubelet.py]: This probably need to be in prow config `--repo=k8s.io/kubernetes=foo`

Comment (krzyzacy) [inline scenarios/kubernetes_kubelet.py]: And may make sure `os.path.basename(cwd) == 'kubernetes'`

Comment (yguo0905) [inline scenarios/kubernetes_kubelet.py]: Done. PTAL!

**Comment by krzyzacy** [comment]
/lgtm

let's try :-)

File: prow/config.yaml [hunk 2]
```diff
@@ -6191,6 +6191,41 @@ periodics:
...
+    - args:
+      - --repo=k8s.io/kubernetes=release-1.7
+      - --root=/go/src
+      - --timeout=70
+      env:
+      - name: GOPATH
+        value: /go
+      - name: GOOGLE_APPLICATION_CREDENTIALS
+        value: /etc/service-account/service-account.json
+      - name: USER
```
Comment (krzyzacy) [inline prow/config.yaml]: I think change this to `GCE_USER`, and remove the `GCE_USER=jenkins` line in the scenario will fix the public key issue.

**Pull request merged**